### DISCUSSION TITLE
Welcome to stopwatch Discussions!

### DISCUSSION THREAD

**Discussion by alexanderism55** [alexanderism55/stopwatch#8] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by alexanderism55** [comment]
# BUG

## there is a bug in the code
The bug is very subtle and it requires some attention to be noticed. what happens is that when you stop counting ard continue counting there is a lag in the counting. The reason for that is that the all the hours, minutes, seconds and milliseconds work independently of each other. They work by the interval, and when you stop counting and continue again the interval is restarted. Hence there is a lag.

**Reply by sammou00** [reply]
Hello, very good job, here are some notes : 
1- `data.js` :  your data always should be an array or an object and not primitive types.
  -  Use `camel-case` to name your variables.
  - why constants.js why not data.js 

2-  you have 4 different set intervals, but you only need one. 

3- your handlers : should be only functions and you should try to avoid using event listener in your handlers 
![image](https://user-images.githubusercontent.com/67711948/185734654-abfc4c7f-5fac-43d7-8f8b-e452649e1abc.png)

 
![image](https://user-images.githubusercontent.com/67711948/185734400-63e5365e-2baf-4a3d-9df9-dcfa8578d4b5.png)

4- in `separation-of-concerns` you have studied => logic function, component functions , handlers and listeners .  It will be a good idea to try to sperate your code according to this 4 categories.

5- `init.js` : you should import your listeners to this file
